src/v/utils/retry_chain_node.h

```diff
@@ -202,14 +202,6 @@ class retry_chain_node {
 
     /// Create a head of the chain without backoff
     retry_chain_node();
-    /// Creates a head with the provided deadline and
-    /// backoff granularity.
-    retry_chain_node(
-      ss::lowres_clock::time_point deadline,
-      ss::lowres_clock::duration initial_backoff);
-    retry_chain_node(
-      ss::lowres_clock::duration timeout,
-      ss::lowres_clock::duration initial_backoff);
```

Comment:
Good call to remove this. Just to clarify why this was added. In situation when the fiber uses some other abort source to stop or doesn't have an abort source at all (which is never the case) it might be useful to have this.